test/unit/StreamrClient.test.js

```diff
@@ -586,6 +589,48 @@ describe('StreamrClient', () => {
         })
     })
 
+    describe('resend()', () => {
+        it('should not send SubscribeRequest on reconnection', async () => {
+            connection.expect(ResendLastRequest.create('stream1', 0, '0', 10, 'session-token'))
+            await client.resend({
+                stream: 'stream1',
+                resend: {
+                    last: 10
+                }
+            }, () => {})
+            await client.pause()
+            await client.connect()
+        })
+        it('should not send SubscribeRequest after ResendResponseNoResend on reconnection', async () => {
+            connection.expect(ResendLastRequest.create('stream1', 0, '0', 10, 'session-token'))
+            const sub = await client.resend({
+                stream: 'stream1',
+                resend: {
+                    last: 10
+                }
+            }, () => {})
+            const resendResponse = ResendResponseNoResend.create(sub.streamId, sub.streamPartition, '0')
+            connection.emitMessage(resendResponse)
+            await client.pause()
+            await client.connect()
+        })
+        it('should not send SubscribeRequest after ResendResponseResent on reconnection', async () => {
+            connection.expect(ResendLastRequest.create('stream1', 0, '0', 10, 'session-token'))
+            const sub = await client.resend({
+                stream: 'stream1',
+                resend: {
+                    last: 10
+                }
+            }, () => {})
+            const msg1 = msg(sub.streamId, {}, '0')
+            connection.emitMessage(msg1)
+            const resendResponse = ResendResponseResent.create(sub.streamId, sub.streamPartition, '0')
+            connection.emitMessage(resendResponse)
+            await client.pause()
+            await client.connect()
+        })
+    })
```

Comment:
A test case is missing perhaps: what happens if the disconnection happens during resend?

- Resend request is sent
- (`ResendResponseResent` is not received)
- disconnect
- reconnect

So what should happen? Well, ideally the resend request might get repeated on reconnect, but that's not super important, it could be handled on application level. Let's leave that out of scope. 

But I guess the client and sub should at least end up in a healthy state. Compared to the situation where the `ResendResponseResent` **is** received, now the `initial_resend_done` event is never fired, and the `sub` is never cleaned up via `this._removeSubscription(sub)`? Not sure if that matters much, I guess it's just a minor memory leak.